docs/reference/fields.txt

```diff
@@ -447,6 +447,81 @@ matches strings containing "hello" before a newline, besides strings ending in
 This is because the meaning of ``$`` is different between PCRE and Ruby
 regular expressions.
 
+
+.. _typed-arrays:
+
+Field Type: Typed Arrays
+------------------------
+
+Mongoid supports declaring a typed array field on models. This field will
+enforce the type of the elements in the array by ``mongoizing`` each element
+inserted into the array. To make a typed array field:
+
+.. code:: ruby
+
+  class Band
+    include Mongoid::Document
+    field :members, type: Array(String)
+  end
+
+The members field on the Band class will have class ``Mongoid::StringArray``.
```

Comment:
One consequence of ``Mongoid::StringArray`` is that the class is reused for multiple models, in other words if an application wanted to customize their typed array for a particular model this wouldn't work well. Do you think this is a valid use case?

Reply:
Can you explain a specific use case for a customization? 

I feel like at some point we have to choose a functionality to provide out of the box, and if the user wants something else they can make a custom class.